crates/sql/src/parser.rs

```diff
@@ -136,29 +137,45 @@ impl<'a> DeltaParser<'a> {
     /// Parse a new expression
     pub fn parse_statement(&mut self) -> Result<Statement, ParserError> {
         match self.parser.peek_token().token {
-            Token::Word(w) => {
-                match w.keyword {
-                    Keyword::VACUUM => {
-                        self.parser.next_token();
-                        self.parse_vacuum()
-                    }
-                    _ => {
-                        // use the native parser
-                        // TODO fix for multiple statememnts and keeping parsers in sync
-                        let stmt = DFParserBuilder::new(self.sql).build()?.parse_statement()?;
-                        self.parser.parse_statement()?;
-                        Ok(Statement::Datafusion(stmt))
-                    }
+            Token::Word(w) => match w.keyword {
+                Keyword::VACUUM => {
+                    self.parser.next_token();
+                    self.parse_vacuum()
                 }
-            }
-            _ => {
-                // use the native parser
-                // TODO fix for multiple statememnts and keeping parsers in sync
-                let stmt = DFParserBuilder::new(self.sql).build()?.parse_statement()?;
-                self.parser.parse_statement()?;
-                Ok(Statement::Datafusion(stmt))
+                _ => self.parse_with_native(),
+            },
+            _ => self.parse_with_native(),
+        }
+    }
+
+    /// Use the native parser to parse the statement
+    // TODO fix for multiple statememnts and keeping parsers in sync
+    fn parse_with_native(&mut self) -> Result<Statement, ParserError> {
+        // DataFusion's parser returns DataFusionError as it may also add
```

Comment:
It is unfortunate to have to translate back from DataFusion error to a ParserError, but I think it is unavoidable as now DataFusion will potentially augment ParserErrors with Span and context information

To retain the same API we need to strip off any additional information and return the underlying `ParserError`